### PR TITLE
 Support installation of Itential artifacts using repository URL

### PR DESCRIPTION
**Pull request by WadeCStern**
Added functionality that will directly move files from the nexus or jfrog repository onto the hosts. Supports either username/password or api keys(jfrog).

### CODE REVIEW

File: roles/platform/tasks/install-platform.yml [hunk 1]
```diff
@@ -1,61 +1,15 @@
...
-    dest: "{{ iap_install_dir }}/{{ iap_bin_file }}"
-    rsync_opts:
-      - "--copy-links"
-  when:
-    - upload_using_rsync
-    - iap_bin_file is defined
+- name: Download archive files local
+  when: iap_bin_file is defined or iap_tar_file is defined
+  ansible.builtin.include_tasks:
+    file: install-platform-from-local.yml
```
Comment (kvelarde-itential) [inline roles/platform/tasks/install-platform.yml:7]:
```suggestion
    file: upload-platform-archive-from-local.yml
```

File: roles/platform/tasks/install-platform.yml [hunk 2]
```diff
@@ -1,61 +1,15 @@
...
-  ansible.builtin.file:
-    path: "{{ iap_install_dir }}/{{ iap_tar_file }}"
-    owner: "{{ iap_user }}"
-    group: "{{ iap_group }}"
-    mode: "0755"
-  when: iap_tar_file is defined
+- name: Download archive files repository
+  when: iap_archive_download_url is defined
+  ansible.builtin.include_tasks:
+    file: install-platform-from-repo.yml
```
Comment (kvelarde-itential) [inline roles/platform/tasks/install-platform.yml:12]:
```suggestion
    file: download-platform-archive-from-repo.yml
```

**Comment by kvelarde-itential** [comment]
This looks really good.  We also need to update the main README.

**Review by kvelarde-itential — CHANGES_REQUESTED** [review]

File: roles/platform/tasks/install-platform-from-repo.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  ansible.builtin.get_url:
+    url: "{{ iap_archive_download_url }}"
+    dest: "{{ iap_install_dir }}/"
+    mode: '0755'
+    headers: >-
+      {%- if repository_api_key is defined and iap_archive_download_url is search("jfrog") -%}
+        {"X-JFrog-Art-Api": "{{ repository_api_key }}", "Accept": "application/octet-stream"}
+      {%- else -%}
+        {"Accept": "application/octet-stream"}
+      {%- endif -%}
```
Comment (steven-schattenberg-itential) [inline roles/platform/tasks/install-platform-from-repo.yml:14]: This probably needs the same comment as the other time we set these headers.  I assume that the else condition is for a Nexus request, but I am assuming.

**Review by kvelarde-itential — APPROVED** [review]

**Pull request merged**